core/trino-main/src/test/java/io/trino/sql/planner/TestLogicalPlanner.java

```diff
@@ -1109,6 +1109,10 @@ public void testRemovesTrivialFilters()
                 "SELECT * FROM nation WHERE 1 = 0",
                 output(
                         values("nationkey", "name", "regionkey", "comment")));
+        assertPlan(
```

Comment:
How does this test correspond to the change? Could you add a test with `WHERE null`?

Reply:
> How does this test correspond to the change?

as a side effect of the fixes, this now works

> Could you add a test with WHERE null?

absolutely!

Reply:
> as a side effect of the fixes, this now works

Could you please explain what changes for this case? I can't see a link, so perhaps I'm reading the code wrong.

Reply:
Now it's clear. Thanks!

Reply:
The appreciation of the clearness is highly appreciated